### DISCUSSION TITLE
Packet Sender in TCP is not a TCP packet sender. It is a TCP client.

### DISCUSSION THREAD

**Discussion by badbeef** [dannagle/PacketSender#242] [General]
I downloaded PacketSender looking for a way to send a raw TCP packet. However, just as I suspected, it does not allow me to send a standalone TCP packet but rather start a new TCP connection and send the TCP payload. Application like this certainly has its uses. But I must say that I am rather disappointed that it is not a true "Packet Sender" as the name suggested, because it comes with the full TCP handshake, which I don't want. In fact, I don't think there is a way to send just an L3 packet. It has to be TCP/UDP (I have not used SSL).

**Comment by dannagle** [comment]
What you are looking for is software developed on top of pcap or npcap (formerly known as WinPcap) if you are on Windows. That will give you low-level access packet crafting. 

More info:
https://nmap.org/npcap/

https://en.wikipedia.org/wiki/Pcap

There are various tutorials and resources out there on how to use the pcap C libraries. It could be just what you need.

Hope that helps.
